registrations/serializers.py

```diff
@@ -330,6 +324,48 @@ def validate(self, data):
         return data
 
 
+class SignUpGroupCreateSerializer(serializers.ModelSerializer, CreateSignUpsSerializer):
+    reservation_code = serializers.CharField(write_only=True)
+
+    def create(self, validated_data):
+        validated_data["created_by"] = self.context["request"].user
+        validated_data["last_modified_by"] = self.context["request"].user
+
+        validated_data.pop("reservation_code")
+        reservation = validated_data.pop("reservation")
+        signups_data = validated_data.pop("signups")
+
+        instance = super().create(validated_data)
+
+        for signup_data in signups_data:
+            signup_data["signup_group"] = instance
+            signup_serializer = SignUpSerializer(data=signup_data, context=self.context)
+            signup_serializer.create(signup_data)
+
+        reservation.delete()
+
+        return instance
+
+    class Meta:
+        fields = (
+            "id",
+            "registration",
+            "signups",
+            "reservation_code",
+            "extra_info",
+        )
+        model = SignUpGroup
+
+
+class SignUpGroupSerializer(serializers.ModelSerializer):
+    view_name = "signupgroup-detail"
+    signups = SignUpSerializer(many=True)
+
+    class Meta:
+        fields = "__all__"
```

Comment:
@tuomas777 raised some concerns on the use of `__all__` without having a test checking which fields are returned by the API. This is to avoid accidental leaks of personal information, if new fields are added later (been there done that ). 

Tuomas requests a test case to verify the returned fields (as opposed to just enumerating the fields with `fields`)

Reply:
See utility method 'assert_fields_exist`

Reply:
Yes, from a security standpoint not only testing for, but also explicitly defining only the fields that can be shown makes a lot of sense. I'll do this change in a moment.

Reply:
@voneiden now changed in https://github.com/City-of-Helsinki/linkedevents/pull/675/commits/ec65bac9ed6106573548849220928f38d6c37d33